crates/matrix-sdk-ui/src/timeline/builder.rs

```diff
@@ -264,6 +264,12 @@ impl TimelineBuilder {
                         RoomEventCacheUpdate::UpdateTimelineEvents { diffs, origin } => {
                             trace!("Received new timeline events diffs");
 
+                            // Adding items coming from paginations or syncs will cause errors
+                            // in the pinned event timeline, so we avoid that here.
+                            if is_pinned_events {
+                                continue;
+                            }
```

Comment:
Having thought about it: i think we want to react to room event cache timeline updates *if and only if* this is a live timeline:

- a pinned events timeline handles its events by itself
- ditto for the permalink timeline
- a future focus like "threads" would likely also manage its events by itself

We need to find a more general solution to this problem, so as to be able to handle sync events live for those timeline focus, but, in the meanwhile, let's disable it entirely with a condition like `if !is_live { continue ; }`, and a comment similar (but more general) to what you've written.

cc @Hywan

Reply:
I'm trying adding the suggested change, but that seems to cause the `timeline::focus_event::test_focused_timeline_reacts` test to never finish. Wouldn't this mean it's breaking reactions in the focused event timeline?

Reply:
yeah, it would be kind of expected too i imagine :/

Reply:
Should I just ignore that test with a `TODO/FIXME`?

Reply:
or also invert its meaning, that live aggregations aren't taken into account on a focused timeline?